src/vs/editor/contrib/referenceSearch/common/referenceSearch.ts

```diff
@@ -39,4 +39,30 @@ export function provideReferences(model: IReadOnlyModel, position: Position, pro
 	});
 }
 
+export function provideExternalReferences(model: IReadOnlyModel, query: ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: IReferenceInformation[]) => void): TPromise<IReferenceInformation[]> {
+
+	// collect references from all providers
+	const promises = ExternalReferenceProviderRegistry.ordered(model).map(provider => {
+		return asWinJsPromise((token) => {
+			return provider.provideExternalReferences(query, hints, token, progress);
+		}).then(result => {
+			if (Array.isArray(result)) {
```

Comment:
This function is copied from `provideReferences` above, which is written by the vs code team. There may be a subtle reason for this, or this may be completely pointless code, I don't really know.

In the interest of getting something working, my strategy is to just copy exactly what vscode has done.